src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -817,9 +817,16 @@ private async Task CommonStreaming(ConnectionState connectionState, string strea
 
         Log.CompletingStream(_logger, streamId);
 
-        // Don't use cancellation token here
-        // this is triggered by a cancellation token to tell the server that the client is done streaming
-        await SendWithLock(connectionState, CompletionMessage.WithError(streamId, responseError), cancellationToken: default).ConfigureAwait(false);
+        try
+        {
+            // Don't use cancellation token here
+            // this is triggered by a cancellation token to tell the server that the client is done streaming
+            await SendWithLock(connectionState, CompletionMessage.WithError(streamId, responseError), cancellationToken: default).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            Log.SendStreamCompletionFailed(_logger, streamId, ex);
```

Comment:
If this fails, does that failure propagate anywhere in the client side API?

Reply:
The logging or the exception we're now catching? The exception wasn't visible anywhere.

Reply:
For consistency, would it make more sense to log this as `Log.ErroredStream` like we do if `SendWithLock` above if `createAndConsumeStream()` throws from `SendWithLock`?

Most of the time if we get here, we're just going to end up logging the same exception twice with different log IDs, right? Getting here means that the connection is dead. And if that's true, that does seem like the likely cause for the streaming failure that just occurred.

Can we check the connection state and make sure we're still connected before trying this?

Reply:
> Most of the time if we get here, we're just going to end up logging the same exception twice

No, most of the time you get here it should be when the client completed the stream or canceled the stream cleanly.

But to your point, if the reason we got here is due to a non-canceled exception, then that would mean the connection was closed and we'd log twice. I'll try to clean that up.